EXPERIMENTS.md

```diff
@@ -58,3 +58,20 @@ Changes the file lock implementation from github.com/nightlyone/lockfile to gith
 When the experiment is enabled the agent will use different lock files from agents where the experiment is disabled, so agents with this experiment enabled should not be run on the same host as agents where the experiment is disabled.
 
 **Status**: Being tested, but it's looking good. We plan to switch lock implementations over the course of a couple of releases, switching over in such a way that nothing gets broken.
+
+### `job-api`
+
+Exposes a local API for the currently running job to introspect and mutate its state in the form of environment variables. This allows you to write scripts, hooks and plugins in languages other than bash, using them to interact with the agent.
```

Comment:
> This allows you to write scripts, hooks and plugins in languages other than bash ...

[🚨 Pedantry mode activated 🚨]

I don't think this is strictly 100% true. The API definitely makes it possible for almost any program running under a hook or plugin script to alter environment variables, but the job executor still assumes that hooks and plugins are _scripts_.

I don't know if it's worth changing the text here, though.

Maybe we should start thinking about the counterpart change (another experiment?) to make the job executor run hooks and plugins directly (without sourcing through the scriptwrapper)? (Or does this exist already?)

Reply:
> Maybe we should start thinking about the [...] change to make the job executor run hooks and plugins directly

I definitely agree that we should start thinking about it but.... i suspect we might not actually be able to do it. some commands are already commands that we wrap in a script (eg, `echo "hello"`), but i suspect we'll probably have to keep the scriptwrapper around for the forseeable future as a backwards-compatibility thing... we have no way of knowing whether something is a script that will use our API, or a script that's expecting to have its env captured, so i suspect we'll need to keep wrapping everything in the scriptwrapper regardless